### ISSUE TITLE
Deconvolution

### ISSUE THREAD

**Issue by bhack** [Shathe/Semantic-Segmentation-Tensorflow-Eager#1]
Are the deconv (transpose conv) going to create artifacts in the decoder?

https://distill.pub/2016/deconv-checkerboard/ (there are many mentions to this 2016 paper)

**Comment by bhack** [comment]
I.e. Check https://arxiv.org/abs/1806.02658

**Comment by Shathe** [comment]
Yes, it could be. I think that's kind of controversial because that post focuses more on RGB images, I mean as a results, where those artifacts matter more than in semantic segmentation.

I see different architectures with upsamplings and with deconvolutions but I do not see any community agreement. 
I edited the code now with the upsampling layer (I did not deleted the traspose convolution code). Nevertheless, Deeplabv3 architecture relies on the upsampling in stead of the convolutions, so I guess as MNasnet is also from Google, I'll go with the upsampling haha

**Comment by Shathe** [comment]
What do you think?

**Comment by bhack** [comment]
If you are interested the mentioned arxiv paper code is available at https://github.com/tensorflow/models/pull/5430
